cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -837,7 +840,9 @@ public MapAndStatus(T loadMap, K status) {
                 return requestRebalance(reconciliation, host, apiClient, kafkaRebalance, true, rebalanceOptionsBuilder);
             default:
                 LOGGER.warnCr(reconciliation, "Ignore annotation {}={}", ANNO_STRIMZI_IO_REBALANCE, rebalanceAnnotation);
-                return Future.succeededFuture(new MapAndStatus<>(null, buildRebalanceStatusFromPreviousStatus(kafkaRebalance.getStatus(), validate(reconciliation, kafkaRebalance))));
+                Set<Condition> conditions = validate(reconciliation, kafkaRebalance);
```

Comment:
What does the `validate` method do? Shouldn't it validate the annotation and set the condition?

Reply:
The `validate` method is used to check for deprecated and unknown properties in the CR